cumulus/client/cirrus-executor/src/aux_schema.rs

```diff
@@ -47,18 +59,28 @@ pub(super) fn write_execution_receipt<Backend: AuxStore, Block: BlockT>(
 		new_first_saved_receipt = block_number.saturating_sub((PRUNING_DEPTH - 1).saturated_into());
 
 		let mut keys_to_delete = vec![];
-		let mut to_delete_start = first_saved_receipt;
-		while to_delete_start < new_first_saved_receipt {
-			keys_to_delete.push((EXECUTION_RECEIPT_KEY, to_delete_start).encode());
-			to_delete_start = to_delete_start.saturating_add(One::one());
+		let mut to_delete_block_number = first_saved_receipt;
+		while to_delete_block_number < new_first_saved_receipt {
```

Comment:
`to_delete_block_number` is basically `from` and `new_first_saved_receipt` is basically `to`.

Instead of transforming those into `Block::Header::Number`, I'd convert them into `subspace_core_primitives::BlockNumber` and iterated over the range of numbers instead, that would be easier to read rather than manually incrementing stuff.

Reply:
Converting to `subspace_core_primitives::BlockNumber` might be helpful to simplify the loop, but you have to convert it back to `Number` as the encode for `u32` and `Block::Header::Number` is different(discovered by the existing test, perhaps  the compact of codec is related, I didn't look into this) unless you want to use `u32` as the part of the final key to insert into the store.

Reply:
I think it is fine to use `BlockNumber` as a final key. Also it might be different in tests (I see that often tests use `u64` for block number), but should be the same in production since we use `u32` everywhere as a block number otherwise.